addons/udes_stock/models/stock_picking.py

```diff
@@ -281,18 +281,45 @@ def _compute_picking_packages(self):
     )
     def _compute_related_picking_ids(self):
         Picking = self.env["stock.picking"]
-        for picking in self:
-            if picking.id:
-                back_orders = Picking.get_pickings(backorder_id=picking.id)
-                picking.u_created_back_orders = back_orders
-                picking.u_back_orders_qty = len(back_orders)
-
-            next_pickings = picking.mapped("move_lines.move_dest_ids.picking_id")
-            prev_pickings = picking.mapped("move_lines.move_orig_ids.picking_id")
-            picking.u_prev_picking_ids = prev_pickings
-            picking.u_prev_pickings_qty = len(prev_pickings)
-            picking.u_next_picking_ids = next_pickings
-            picking.u_next_pickings_qty = len(next_pickings)
+        if self.ids:
+            # To avoid messy ORM joins, get next/prev picking ids
+            # with a direct query for all picks in self
+            self.env.cr.execute(
+                """
+                SELECT
+                    sp.id,
+                    array_agg(DISTINCT next_pick.id) AS next_picks,
+                    array_agg(DISTINCT prev_pick.id) AS prev_picks,
+                    array_agg(DISTINCT backorder.id) AS backorder_picks
+                FROM
+                    stock_picking sp
+                    LEFT JOIN stock_picking backorder ON backorder.backorder_id = sp.id
+                    LEFT JOIN stock_move sm ON sm.picking_id = sp.id
+                    LEFT JOIN stock_move_move_rel dest_move_rel ON sm.id = dest_move_rel.move_orig_id
+                    LEFT JOIN stock_move_move_rel orig_move_rel ON sm.id = orig_move_rel.move_dest_id
+                    LEFT JOIN stock_move orig_move ON orig_move.id = orig_move_rel.move_orig_id
+                    LEFT JOIN stock_move dest_move ON dest_move.id = dest_move_rel.move_dest_id
+                    LEFT JOIN stock_picking next_pick ON dest_move.picking_id = next_pick.id
+                    LEFT JOIN stock_picking prev_pick ON orig_move.picking_id = prev_pick.id
+                WHERE
+                    sp.id IN %s
+                GROUP BY sp.id
+            """,
+                (tuple(self.ids),),
```

Comment:
If length is 1 i think this will raise sql syntax error:

u think should be an extra check of the length of self.ids
'if len(self.ids) == 1:
   pickings = "(f{self.ids[0]})"
else:
   pickings = tuple(self.ids)'

And replace on query with pickings variable. Have a look though if i am right

Reply:
`self.ids` will always return a list, so it should be fine I would have thought.

But yes worth checking out to be safe... speeding up is good, but not so good when it comes with an exception. :grin:

Reply:
Ah I see. Well if we load a picking in the form view then `self` would only have one record anyway right?

@caleb-shelton please check this, but hopefully fine. :)

Reply:
It works fine with self having one record. Also the SQL is checking 'IN'